src/cargo/ops/vendor.rs

```diff
@@ -207,8 +207,10 @@ fn sync(
         let dst = canonical_destination.join(&dst_name);
         to_remove.remove(&dst);
         let cksum = dst.join(".cargo-checksum.json");
-        if dir_has_version_suffix && cksum.exists() {
-            // Always re-copy directory without version suffix in case the version changed
+        // The standard registry is the only immutable source,
+        // path and git dependencies' versions cannot be trusted to mean "no change"
+        if dir_has_version_suffix && id.source_id().is_crates_io() && cksum.exists() {
```

Comment:
Will this have the effect of *always* revendoring git-sourced packages even if they haven't changed?

Reply:
Yes. Unfortunately in `.cargo-checksum.json` there is no checksum recorded for git dependencies.

Reply:
I think there is no way currently to retrieve the git rev of the vendored dependency.